### PR TITLE
SelectBox: customItemCreating should not be called more then once on one input (T893205)

### PR DESCRIPTION
**Pull request by ksercs**

### CODE REVIEW

**Review by DokaRus — DISMISSED** [review]

File: js/ui/select_box.js [hunk 1]
```diff
@@ -673,8 +673,10 @@ const SelectBox = DropDownList.inherit({
     },
 
     _valueChangeEventHandler: function(e) {
-        if(this.option('acceptCustomValue') && this._isCustomItemSelected()) {
+        if(this.option('acceptCustomValue') && this._isCustomItemSelected() && !this.isValueChanging) {
+            this.isValueChanging = true;
             this._customItemAddedHandler(e);
+            this.isValueChanging = false;
```
Comment (San4es) [inline js/ui/select_box.js]:
```suggestion
        if(this.option('acceptCustomValue') && this._isCustomItemSelected() && !this._isValueChanging) {
            this._isValueChanging = true;
            this._customItemAddedHandler(e);
            this._isValueChanging = false;
```

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 2]
```diff
@@ -2221,6 +2221,30 @@ QUnit.module('editing', moduleSetup, () => {
...
+            onCustomItemCreating: (e) => {
+                ++handlerCallCount;
+                handlerCallCount === 1 && $selectBox.dxSelectBox('instance')._valueChangeEventHandler();
+                e.customItem = null;
+            }
+        });
+        const $input = $selectBox.find(toSelector(TEXTEDITOR_INPUT_CLASS));
+        const keyboard = keyboardMock($input);
+
+        $input.focus();
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Can we use `keyboard.focus()` here?

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 3]
```diff
@@ -2221,6 +2221,30 @@ QUnit.module('editing', moduleSetup, () => {
...
+    QUnit.test('onCustomItemCreating should not be called twice when there is blur in the handler (T893205)', function(assert) {
+        let handlerCallCount = 0;
+
+        const $selectBox = $('#selectBox').dxSelectBox({
+            items: ['1', '2', '3'],
+            opened: true,
+            acceptCustomValue: true,
+            onCustomItemCreating: (e) => {
+                ++handlerCallCount;
+                handlerCallCount === 1 && $selectBox.dxSelectBox('instance')._valueChangeEventHandler();
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]:
```suggestion
                if(handlerCallCount === 1) {
                    e.component._valueChangeEventHandler();
                }
```

Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js:2233]: Can we avoid using this private method?

File: js/ui/select_box.js (hunk 1 above)
Comment (ksercs) [reply js/ui/select_box.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js (hunk 2 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js (hunk 3 above)
Comment (ksercs) [reply testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: fixed

File: testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js [hunk 4]
```diff
@@ -2221,6 +2221,31 @@ QUnit.module('editing', moduleSetup, () => {
...
+        let handlerCallCount = 0;
+
+        const $selectBox = $('#selectBox').dxSelectBox({
+            items: ['1', '2', '3'],
+            opened: true,
+            acceptCustomValue: true,
+            onCustomItemCreating: (e) => {
+                ++handlerCallCount;
+                if(handlerCallCount === 1) {
+                    $selectBox.dxSelectBox('instance')._valueChangeEventHandler();
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/selectBox.tests.js]: Can we avoid using this private method?

**Review by San4es — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**